react-carousel-component/my-app/src/RotatingBanner.jsx

```diff
@@ -0,0 +1,49 @@
+import './styles.css';
+import Banner from './Banner';
+import PrevButton from './PrevButton';
+import NextButton from './NextButton';
+import Indicators from './Indicators';
+import { useState, useEffect } from 'react';
+
+type props = {
+  items: string[],
+};
+
+export default function RotatingBanner({ items }: props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setInterval(
+      () => setCurrentIndex((currentIndex + 1) % items.length),
+      2000
+    );
+    return () => clearInterval(timerId);
+  }, [currentIndex, items.length]);
```

Comment:
I always debate on these: Do I make the dependency `items.length` or just`items`. I think the answer comes from the question, "What would happen if I got new items but the same number of them so .length doesn't change?"

In this case (but certainly not in all cases), it looks like it would continue operating correctly, so items.length is the right answer (just like you have).